UnitySDK/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -273,67 +267,59 @@ private void InitializeEnvironment()
             // If it fails, we check if there are any external brains in the scene
             // and if Unity is in Editor mode 
             // If there are : Launch the communicator on the default port
-            // If there are not, there is no need for a communicator and it is set
-            // to null
             catch
             {
-                communicator = null;
 #if UNITY_EDITOR
-                if (controlledBrains.ToList().Count > 0)
+                m_Communicator = null;
+                if (controlledBrains.Any())
                 {
-                    communicator = new RpcCommunicator(
-                        new CommunicatorParameters
+                    m_Communicator = new RpcCommunicator(
+                        new CommunicatorInitParameters
                         {
                             port = 5005
                         });
                 }
 #endif
             }
-            m_BrainBatcher = new Batcher(communicator);
-            if (communicator != null)
+            foreach (var trainingBrain in controlledBrains)
             {
+                trainingBrain.SetCommunicator(m_Communicator);
+            }
 
-                foreach (var trainingBrain in controlledBrains)
-                {
-                    trainingBrain.SetBatcher(m_BrainBatcher);
-                }
-                m_IsCommunicatorOn = true;
-
-                var academyParameters =
-                    new CommunicatorObjects.UnityRLInitializationOutputProto();
-                academyParameters.Name = gameObject.name;
-                academyParameters.Version = k_ApiVersion;
-                foreach (var brain in controlledBrains)
-                {
-                    var bp = brain.brainParameters;
-                    academyParameters.BrainParameters.Add(
-                        bp.ToProto(brain.name, true));
-                }
+            if (m_Communicator != null)
+            {
+                m_Communicator.QuitCommandReceived += OnQuitCommandReceived;
+                m_Communicator.ResetCommandReceived += UpdateResetParameters;
+                m_Communicator.RLInputReceived += OnRLInputReceived; 
+                
 
-                academyParameters.EnvironmentParameters =
-                    new CommunicatorObjects.EnvironmentParametersProto();
-                foreach (var key in resetParameters.Keys)
-                {
-                    academyParameters.EnvironmentParameters.FloatParameters.Add(
-                        key, resetParameters[key]
-                    );
-                }
+                m_IsCommunicatorOn = true;
```

Comment:
Can we get rid of this and just `m_Communicator != null` (or make it a property)?

Reply:
done